src/runtime.ts

```diff
@@ -631,6 +586,143 @@ export async function createDefaultRuntime(
+      // eslint-disable-next-line no-param-reassign
+      (runtimeConfig as any).accountMap = {};
+    }
+    if (!runtimeConfig.keyConfig) {
+      // eslint-disable-next-line no-param-reassign
+      (runtimeConfig as any).keyConfig = {};
+    }
+    Object.assign(runtimeConfig.accountMap, tempConfig.accountMap);
+    // do not apply the encryption key directly, will be handled by keyConfig account + password
+    // handling
+    Object.assign(runtimeConfig.keyConfig, {
+      [runtimeConfig.account]: runtimeConfig.password,
+    });
+    // cleanup runtime
+    delete runtimeConfig.mnemonic;
+    delete runtimeConfig.password;
+  } else if (!runtimeConfig.accountMap
+       || !(Object.keys(runtimeConfig.accountMap).length)) {
+    throw new Error('accountMap invalid');
+  }
+
+  // check and modify if any accountid with password is provided
+  if (runtimeConfig.keyConfig) {
+    const accountIds = Object.keys(runtimeConfig.keyConfig);
+    await Promise.all(accountIds.map(async (accountId: string) => {
+      // check if the key is a valid accountId
+      if (accountId.length === 42) {
+        let dataKey;
+        let useDefaultDatakey = !runtimeConfig.useIdentity;
+        const sha3Account = web3.utils.soliditySha3(accountId);
+        const sha9Account = web3.utils.soliditySha3(sha3Account, sha3Account);
+
+        // if useIdentity is specified, try to generate dataKeys with identity
+        if (runtimeConfig.useIdentity) {
+          if (!contextLessRuntime) {
+            // create a runtime without account relation, just for accesing some bcc class instances
+            contextLessRuntime = createRuntime(
+              web3,
+              dfs,
+              {
+                accountMap: {
+                  [nullAddress]: 'no-private-key-here',
+                },
+              },
+              options,
+            );
+          }
+
+          // try out several dataKey encryption salting methods and check if, generated dataKeys are
+          // correct
+          let identity = nullAddress;
+          try {
+            identity = await (await contextLessRuntime).verifications
+              .getIdentityForAccount(accountId, true);
+          } catch (ex) {
+            // when no identity for a account could be found, it's a uninitialized account, so we
+            // should use default data key
+            useDefaultDatakey = true;
+            logger.log(`Could not find identity for account ${accountId} during keyConfig resolve.`
+              + ' Using default data key.', 'warning');
+          }
+
+          if (identity !== nullAddress) {
+            const encryptionSalts = [identity, accountId];
+            // eslint-disable-next-line guard-for-in
+            for (const encryptionSalt of encryptionSalts) {
+              const saltedDataKey = web3.utils
+                .keccak256(encryptionSalt + runtimeConfig.keyConfig[accountId])
+                .replace(/0x/g, '');
```

Comment:
Shouldn't it be enough to replace '0x' only at start of string?

```suggestion
                .replace(/^0x/, '');
```